client/src/main/java/io/pravega/client/segment/impl/SegmentOutputStreamImpl.java

```diff
@@ -379,7 +379,7 @@ public void appendSetup(AppendSetup appendSetup) {
                 log.warn("Connection setup could not be completed because connection is already failed for writer {}", writerId);
                 return;
             }
-            if (toRetransmit == null || toRetransmit.isEmpty()) {
+            if (toRetransmit.isEmpty()) {
```

Comment:
Can `toRetransmit` never be null?

Reply:
yes, to Retransmit can never be null.
(`.collect(Collectors.toList());` in line 376 ensures that)